cpp/lib/src/Elasticsearch.cc

```diff
@@ -18,235 +19,58 @@
-}
-
-
-std::shared_ptr<JSON::ObjectNode> FieldsToJSON(const std::unordered_map<std::string, std::string> &fields) {
-    std::shared_ptr<JSON::ObjectNode> tree_root(new JSON::ObjectNode);
-    for (const auto &field : fields) {
-        std::shared_ptr<JSON::StringNode> value_node(new JSON::StringNode(JSON::EscapeString(field.second)));
-        tree_root->insert(field.first, value_node);
-    }
-    return tree_root;
+    *host                    = ini_file.getString("Elasticsearch", "host");
+    *index                   = ini_file.getString("Elasticsearch", "index");
+    *document_type           = ini_file.getString("Elasticsearch", "document_type");
+    *username                = ini_file.getString("Elasticsearch", "username", "");
+    *password                = ini_file.getString("Elasticsearch", "password", "");
+    *ignore_ssl_certificates = ini_file.getBool("Elasticsearch", "ignore_ssl_certificates", false);
 }
 
 
-std::unordered_map<std::string, std::string> JSONToFields(const std::shared_ptr<const JSON::ObjectNode> &json_object) {
-    std::unordered_map<std::string, std::string> fields;
-    for (const auto &key_and_value : *json_object) {
-        std::shared_ptr<JSON::StringNode> string_node(JSON::JSONNode::CastToStringNodeOrDie(key_and_value.first, key_and_value.second));
-        fields[key_and_value.first] = string_node->getValue();
-    }
-    return fields;
+Elasticsearch::Elasticsearch(const std::string &ini_file_path) {
+    LoadIniParameters(ini_file_path.empty() ? DEFAULT_CONFIG_FILE_PATH : ini_file_path, &host_, &index_, &document_type_, &username_,
+                      &password_, &ignore_ssl_certificates_);
 }
 
 
-/** \brief Sends REST queries to Elasticsearch server
-*   \throws std::runtime_error on REST error or if the JSON response contained an error flag.
-*/
-std::shared_ptr<JSON::ObjectNode> Connection::query(const std::string &action, const REST::QueryType query_type,
-                                                    const std::shared_ptr<const JSON::JSONNode> &data)
-{
-    const Url url(host_.toString() + "/" + action);
-    Downloader::Params params;
-    params.authentication_username_ = username_;
-    params.authentication_password_ = password_;
-    params.ignore_ssl_certificates_ = ignore_ssl_certificates_;
+void Elasticsearch::insertDocument(const std::string &document_id, const std::vector<std::string> &document_chunks) {
+    Downloader::Params downloader_params;
+    downloader_params.authentication_username_ = username_;
+    downloader_params.authentication_password_ = password_;
+    downloader_params.ignore_ssl_certificates_ = ignore_ssl_certificates_;
+    downloader_params.additional_headers_.push_back("Content-Type: application/json");
 
-    if (data != nullptr) {
-        params.additional_headers_.push_back("Content-Type: application/json");
-        LOG_DEBUG(data->toString());
-    }
+    const Url delete_url(host_ + "/" + index_ + "/_delete_by_query");
```

Comment:
If I recall correctly, there should be an update document endpoint as well. We could use that in lieu of deleting the existing entry and reinserting it. Will probably be faster.